File: frontend-dev/src/types/aiHistory.ts
export type AIHistoryItemDTO = {
  id: number;
  created_at: string | null;
  actor_id: number | null;
  actor_name?: string | null;
  action: string;
  entity: string;
  entity_id?: string | null;
  diff_json?: Record<string, unknown> | null;
  request_id?: string | null;
};

export type AIHistoryResponseDTO = {
  items: AIHistoryItemDTO[];
  limit: number;
  offset: number;
  total: number;
};

File: app/schemas/ai_history.py
"""AI history schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AIHistoryItem(BaseModel):
    id: int
    created_at: datetime | None
    actor_id: int | None
    actor_name: str | None
    action: str
    entity: str
    entity_id: str | None
    diff_json: dict[str, Any] | None
    request_id: str | None


class AIHistoryResponse(BaseModel):
    items: list[AIHistoryItem]
    limit: int
    offset: int
    total: int
